middleware_grip.go

```diff
@@ -36,34 +36,40 @@ func setServiceLogger(r *http.Request, logger grip.Journaler) *http.Request {
 	return r.WithContext(context.WithValue(r.Context(), loggerKey, logger))
 }
 
-type logAnnotation struct {
-	key   string
-	value interface{}
-}
+type loggingAnnotations map[string]interface{}
 
 // AddLoggingAnnotation adds a key-value pair to be added to logging
-// messages used by the application logging information. There can be
-// only one annotation registered per-request.
-func AddLoggingAnnotation(r *http.Request, key string, data interface{}) *http.Request {
-	return r.WithContext(context.WithValue(r.Context(), loggingAnnotationsKey, &logAnnotation{key: key, value: data}))
+// messages used by the application logging information, overwriting
+// any previous value for the key.
+// This will noop if the logger hasn't already been set up.
+func AddLoggingAnnotation(r *http.Request, key string, value interface{}) {
+	annotations := getLoggingAnnotations(r.Context())
+	if annotations == nil {
+		return
+	}
+
+	annotations[key] = value
```

Comment:
Does this only affect the local copy of annotations? I think we would need to re-set it now, right? I see the test works though so 🤷

Reply:
I believe the reason it works is that the map is a reference type. Although the _variable_ will be different for each call, the underlying map will always be the same.